plugin/kubernetes/controller.go

```diff
@@ -116,6 +124,17 @@ func newdnsController(ctx context.Context, kubeClient kubernetes.Interface, opts
 		object.DefaultProcessor(object.ToService(opts.skipAPIObjectsCleanup)),
 	)
 
+	dns.ingLister, dns.ingController = object.NewIndexerInformer(
```

Comment:
This informer must be conditionally created, like the pod informer.
When k8s_ingress is not in use, kubernetes should not establish a watch on ingress objects.